### DISCUSSION TITLE
New Functionality After EMTG v9.02

### DISCUSSION THREAD

**Discussion by ednasa** [nasa/EMTG#4] [General]
After EMTG v9.02 here are some of the features the community can look forward to:
- Open source Interior Point OPTimizer (IPOPT) compatibility as an alternative to SNOPT
- GUI Input Validation
- Adaptive Step Integration
- GUI tooltips linked to the User Guide
- Testatron (test bed) updates
- Solar Sail functionality
- Consolidating hardware & universe information into JSON libraries

**Comment by am-nasa-gsfc** [comment]
@ednasa, hope you don't mind me adding the JSON update to your list here. Figured this post will become a running list as we update our work plan.
